modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -410,7 +410,7 @@ private void processJSONPayload(MessageContext synCtx, ScriptMessageContext scri
                     jsonObject = this.jsEngine.eval('(' + scriptWithJsonParser + ')');
                 }
             } catch (ScriptException e) {
-                throw new SynapseException("Invalid JSON payload", e);
+                throw new ScriptException("Invalid JSON payload");
```

Comment:
shall we add exception object as follows:
throw new ScriptException("Invalid JSON payload",e);
Otherise upstream will lose stacktrace

Reply:
Hi,
Since we are throwing ScriptException the constructor (message, e) doesn't exists what we can propagate is the message , filename , linenumber etc.

Reply:
Then please include the filename and line number inside the error message.